Blog recipes for success (#2447)
block/goose: documentation/blog/2025-05-06-recipe-for-success.md/index.md

    - rizel
---
![blog cover](cookingwithgoose.png)

# A Recipe for Success: Cooking Up Repeatable Agentic Workflows

Ratatouille isn't just a heartwarming (and slightly unhygienic) film about a rat chef. It's also analogous to a popular tech trend: AI agents and the [Model Context Protocol (MCP)](https://modelcontextprotocol.io/).

<!-- truncate -->

---

## The Remy-Linguini Dynamic

If you haven't seen the movie, here's the gist: Remy is an incredible chef with all the know-how, but he's a rat, so no kitchen access. Linguini is a kitchen worker with full access but little cooking skill. Together, they form a symbiotic relationship: Remy hides under Linguini's hat and guides him on what tools to use and when. 

If a customer orders fries, Linguini might freeze, but Remy scopes out the kitchen, sees what's available, and gives step-by-step instructions: 

> _"Grab a knife and a cutting board. Now slice the potato."_ 

Then, Linguini executes the plan.

---

## Traditional AI Agents

Agentic systems work similarly. You have three core components:

* A Large Language Model (LLM)  
* An Agent  
* Tools

> The LLM is like Remy; it is full of knowledge and reasoning, but has no hands-on access. The agent is like Linguini; it can take action, but doesn't know what to do unless guided.

Reviewer: ```suggestion
The LLM is like Remy; it is full of knowledge and reasoning, but has no hands-on access. The agent is like Linguini; it can take action, but needs guidance.
```

* Where tools live
* How stations are organized
* How to move efficiently through space
* When to pivot if something's missing

With Colette's guidance, Linguini understands the kitchen as a system. When a customer orders spaghetti, Remy quickly forms a plan: 

> _"Boil the pasta, sauté the garlic and tomatoes, plate it with basil."_

Instead of mindlessly following orders, Linguini is equipped to orchestrate the entire operation by:

* Heading to the pasta station to get water boiling
* Checking the sauté station for clean pans and fresh garlic
* Grabbing the right tools: colander, ladle, sauté pan
* Finding spare pans or changing flow when needed
* Managing ingredients and backup supplies
* Coordinating timing so everything finishes in sync
* Plating and delivering dishes confidently

---

## Built Different

That's how it works with AI agents that follow the Model Context Protocol(MCP). MCP shifts the agent from passive executor to active orchestrator, making it less reliant on an LLM and more aware of the context in which it's operating.

> [Codename Goose](/) is a local, open source AI agent that follows the structure of MCP.

Reviewer: ```suggestion
[Goose](/) is a local, open source AI agent that follows the structure of MCP.
```
